apps/src/lab2/responseValidators.ts

```diff
@@ -118,16 +118,16 @@ export class ValidationError extends Error {
 }
 
 function sourceValidatorHelper(
-  response: Record<string, unknown>,
+  response: Record<string, unknown> | unknown[],
   appSpecificValidator: (response: Record<string, unknown>) => void
 ): ProjectSources {
-  if (!response.source) {
-    throwMissingFieldError('source');
+  if (Array.isArray(response) || !response.source) {
+    throw missingFieldError('source');
   }
   appSpecificValidator(response);
   return response as unknown as ProjectSources;
 }
 
-function throwMissingFieldError(fieldName: string) {
-  throw new ValidationError('Missing required field: ' + fieldName);
+function missingFieldError(fieldName: string) {
+  return new ValidationError('Missing required field :' + fieldName);
```

Comment:
This is just so that the actual `throw` happens in the functions themselves, for TS type narrowing purposes.